cmd/ketch/pool_remove.go

```diff
@@ -35,14 +43,84 @@ type poolRemoveOptions struct {
 }
 
 func poolRemove(ctx context.Context, cfg config, options poolRemoveOptions, out io.Writer) error {
-	pool := ketchv1.Pool{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: options.Name,
-		},
+	var pool ketchv1.Pool
+
+	if err := cfg.Client().Get(ctx, types.NamespacedName{Name: options.Name}, &pool); err != nil {
+		return fmt.Errorf("failed to get pool: %w", err)
 	}
+
+	if userWantsToRemoveNamespace(pool.Spec.NamespaceName, out) {
+		if err := namespaceHasAdditionalResources(ctx, cfg, &pool); err != nil {
+			printNsRemovalErr(out, err)
+		} else {
+			if err := removeNamespace(ctx, cfg, &pool); err != nil {
+				printNsRemovalErr(out, err)
+			} else {
+				fmt.Fprintln(out, "Namespace successfully removed!")
+			}
+		}
+	}
+
 	if err := cfg.Client().Delete(ctx, &pool); err != nil {
 		return fmt.Errorf("failed to remove the pool: %w", err)
 	}
-	fmt.Fprintln(out, "Successfully removed!")
+
+	fmt.Fprintln(out, "Pool successfully removed!")
+
+	return nil
+}
+
+func userWantsToRemoveNamespace(ns string, out io.Writer) bool {
+	response := promptToRemoveNamespace(ns, out)
+	return handleNamespaceRemovalResponse(response, ns, out)
+}
+
+func promptToRemoveNamespace(ns string, out io.Writer) string {
+	fmt.Fprintf(out, "Do you want to remove the namespace along with the pool? Please enter namespace to confirm (%s): ", ns)
+
+	var response string
+	fmt.Scanln(&response)
+
+	return response
+}
+
+func handleNamespaceRemovalResponse(response, ns string, out io.Writer) bool {
+	if response != ns {
+		fmt.Fprintln(out, skipNsRemovalMsg)
+		return false
+	}
+
+	return true
+}
+
+func namespaceHasAdditionalResources(ctx context.Context, cfg config, targetPool *ketchv1.Pool) error {
```

Comment:
`namespaceHasAdditionalPools` ?

Reply:
also, the function name sounds like it will return a bool. I would do something like `checkNamespaceAdditionalPools` or similar.

Reply:
Sure